sklearn_numba_dpex/common/kernels.py

```diff
@@ -330,24 +351,28 @@ def make_sum_reduction_2d_kernel(
     # TODO: at some point, the cost of scheduling the kernel is more than the cost of
     # running the reduction iteration. At this point the loop should stop and then a
     # single work item should iterates one time on the remaining values to finish the
-    # reduction ?
+    # reduction?
     kernel = fused_func_kernel
-    while sum_axis_size > 1:
-        sum_axis_size = math.ceil(sum_axis_size / reduction_block_size)
+    sum_axis_size = size0 if axis == 0 else size1
+    next_input_size = sum_axis_size
```

Comment:
Not that introducing a new `next_input_size` to be updated in the loop instead of updating `sum_axis_size` is both helpful to understand the code but also to avoid introducing potential bugs in subsequent code such as the `sum_reductions` local function. At the moment it was only checking for `if sum_axis_size == 0:` which is a special case that was not impacted but if this code evolves it could have introduced weird bugs.